baseapp/abci.go

```diff
@@ -232,16 +232,40 @@ func (app *BaseApp) EndBlock(req abci.RequestEndBlock) (res abci.ResponseEndBloc
 	return res
 }
 
-// PrepareProposal implements the ability for the application to verify and/or modify transactions in a block proposal.
+// PrepareProposal implements the PrepareProposal ABCI method and returns a
+// ResponsePrepareProposal object to the client. The PrepareProposal method is
+// responsible for allowing the block proposer to perform application-dependent
+// work in a block before proposing it.
+//
+// Transactions can be modified, removed, or added by the application. Since the
+// application maintains it's own local mempool, it will ignore the transactions
+// provided to it in RequestPrepareProposal. Instead, it will determine which
+// transactions to return based on the mempool's semantics and the MaxTxBytes
+// provided by the client's request.
+//
+// Note, there is no need to execute the transactions for validity as they have
+// already passed CheckTx.
+//
+// Ref: https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-060-abci-1.0.md
+// Ref: https://github.com/tendermint/tendermint/blob/main/spec/abci/abci%2B%2B_basic_concepts.md
 func (app *BaseApp) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
-	// treated as a noop until app side mempool is implemented
-	return abci.ResponsePrepareProposal{Txs: req.Txs}
+	panic("PrepareProposal currently not supported")
```

Comment:
note, there is no need to have empty stubs w/o implementing it. Panics will ensure things are properly setup and tested.